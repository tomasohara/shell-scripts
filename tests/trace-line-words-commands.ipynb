{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6dcea75d",
   "metadata": {},
   "outputs": [],
   "source": [
    "## SETUP (OPTIONAL - USE IF REQUIRED)\n",
    "## Bracketed Paste is disabled to prevent characters after output\n",
    "## Example: \n",
    "## $ echo 'Hii'\n",
    "## | Hi?2004l\n",
    "# bind 'set enable-bracketed-paste off'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "65d43538",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PROMPT_MOMQYZNZZMVP\\[\\]>\n"
     ]
    }
   ],
   "source": [
    "## TEST:  Make sure simple prompt used (e.g., no escapes that might contaminate output)\n",
    "## PS1=\"$ \"\n",
    "## TODO: PS1=\"> \"\n",
    "\n",
    "## NOTE: The Jupyter bash kernel requires that PS1 not be modified as they customize it. \n",
    "echo $PS1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d623c9a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "## BAD\n",
    "# # Delete all aliases and function\n",
    "# # TODO: Instead start from pristine environment\n",
    "# unalias -a\n",
    "# alias | wc -l\n",
    "# for f in $(typeset -f | egrep '^\\w+'); do unset -f $f; done\n",
    "# typeset -f | egrep '^\\w+' | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0ff8f157",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Setting a temp directory for tests\n",
    "## OLD: TMP=/tmp/test-linewords\n",
    "\n",
    "## NOTE: Source it directly from the ./tests directory.\n",
    "## OLD: BIN_DIR=$PWD/.."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ab954c7f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "renamed '/tmp/test-trace-line-words/trace-line-words-commands' -> '/tmp/test-trace-line-words/trace-line-words-commands.16Apr24'\n"
     ]
    }
   ],
   "source": [
    "## NOTE: For reproducability, the directory name needs to be fixed\n",
    "## In place of $$, use a psuedo random number (e,g., 6439)\n",
    "## *** All output from one run to the next needs to be the same ***\n",
    "\n",
    "## OLD: Used the concept of trash-dir, instead use unique temp_dirs\n",
    "## temp_dir=$TMP/test-$$\n",
    "# temp_dir=$TMP/test-6439\n",
    "# trash_dir=$TMP/\"_temp-trash-$(date \"+%Y%m%d%H%M%S\")\"\n",
    "# command mkdir -p \"$trash_dir\"\n",
    "# command mkdir -p \"$temp_dir\"\n",
    "# command cd \"$temp_dir\"\n",
    "\n",
    "TMP=${TMP:-/tmp}\n",
    "if [ \"$DEBUG_LEVEL\" -lt 4 ]; then TMP=/tmp/test-trace-line-words; fi\n",
    "temp_dir=$TMP/trace-line-words-commands\n",
    "rename-with-file-date \"$temp_dir\"\n",
    "command mkdir -p \"$temp_dir\"\n",
    "command cd $temp_dir\n",
    "\n",
    "## OLD\n",
    "# mkdir -p \"$temp_dir\"\n",
    "# # TODO: /bin/rm -rvf \"$temp_dir\"\n",
    "# cd \"$temp_dir\"\n",
    "# pwd\n",
    "\n",
    "# #ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)\n",
    "# alias linebr=\"printf '%*s\\n' \"${COLUMNS:-$(tput cols)}\" '' | tr ' ' -\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "b34ce805",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Global Setup\n",
    "alias testuser=\"sed -r \"s/\"$USER\"+/userxf333/g\"\"\n",
    "# alias testnum=\"sed -r \"s/[0-9]/N/g\"\"  \n",
    "## For number of digits not being critical\n",
    "alias testnum=\"sed -r \"s/[0-9][0-9]*/N/g\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ec6df159",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N\n"
     ]
    }
   ],
   "source": [
    "# Count aliases proper\n",
    "alias | wc -l | testnum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "aa40fd1c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N\n"
     ]
    }
   ],
   "source": [
    "# Count functions\n",
    "typeset -f | egrep '^\\w+' | wc -l | testnum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "1a81a0db",
   "metadata": {},
   "outputs": [],
   "source": [
    "## BAD\n",
    "# source $BIN_DIR/tomohara-aliases.bash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "bbcd9a49",
   "metadata": {},
   "outputs": [],
   "source": [
    "## OLD & BAD\n",
    "# # GENERATES A PARAGRAPH WITH MULTIPLE LINES\n",
    "# rm -rf ./*\n",
    "# printf \"Hi Mom,\\nThis is the use of printf\\nI can use a backslash n to start a new line\\n1\\n2\\n3\" >> abc-test.txt\n",
    "# printf \"This is another-test file\" >> test2.txt\n",
    "# printf \"This is test-file 3\" >> test3.txt\n",
    "# printf \"This is a test-file 4\" >> test4.txt\n",
    "\n",
    "printf \"Hi Mom,\\nThis is the use of printf\\nI can use a backslash n to start a new line\\n1\\n2\\n3\" > abc-test.txt\n",
    "printf \"This is another-test file\" > test2.txt\n",
    "printf \"This is test-file 3\" > test3.txt\n",
    "printf \"This is a test-file 4\" > test4.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "1f9ba9a1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "txt\t4\n"
     ]
    }
   ],
   "source": [
    "ls | count-it '\\.([^\\.]+)$'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "9f34ee1c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N\tHi Mom,\n",
      "N\tThis is the use of printf\n",
      "N\tI can use a backslash n to start a new line\n",
      "N\tN\n",
      "N\tN\n",
      "N\tN\n"
     ]
    }
   ],
   "source": [
    "# line-wc | COUNTS WORD FOR EACH LINE\n",
    "cat abc-test.txt | line-wc | testnum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "6c6506f8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N\tHi Mom,\n",
      "N\tThis is the use of printf\n",
      "N\tI can use a backslash n to start a new line\n",
      "N\tN\n",
      "N\tN\n",
      "N\tN\n"
     ]
    }
   ],
   "source": [
    "# line-len | COUNTS CHARACTERS FOR EACH LINE\n",
    "cat abc-test.txt | line-len | testnum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "06a0841f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N\tHi Mom,\n",
      "This is the use of printf\n",
      "I can use a backslash n to start a new line\n",
      "N\n",
      "N\n",
      "N\n"
     ]
    }
   ],
   "source": [
    "# para-len | COUNTS CHARACTERS FOR EACH PARAGRAPH\n",
    "cat abc-test.txt | para-len | testnum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "c50a7083",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N\tHi Mom,\n",
      "N\tThis is the use of printf\n",
      "N\tI can use a backslash n to start a new line\n",
      "N\tN\n",
      "N\tN\n",
      "N\tN\n"
     ]
    }
   ],
   "source": [
    "# line-word-len | COUNTS WORDS FOR EACH LINE\n",
    "cat abc-test.txt | line-word-len | testnum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "9c1a26c6",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NotZero\n"
     ]
    }
   ],
   "source": [
    "## PROCESS DOESN'T END FOR ls-line-len\n",
    "## EXAMPLE: \n",
    "## $ls-line-len\n",
    "## |12      abc-test.txt1h=\n",
    "\n",
    "\n",
    "ls-line-len | { [ $(wc -l < /dev/stdin) -ne 0 ] && echo \"NotZero\" || echo \"Zero\"; }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "07e67b85",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ## ERROR-GENERATED : check-class-dist\n",
    "# $ cat abc-test.txt | check-class-dist\n",
    "# | Can't open : No such file or directory at /home/xaea12/bin/count_it.perl line 171.\n",
    "# | WARNING: unexpected distribution for FREQ (all 0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "93c128a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# $ cat abc-test.txt | 2bib\n",
    "# | bash: bibitem2bib: command not found\n",
    "# | cat: write error: Broken pipe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "85a87403",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done\n"
     ]
    }
   ],
   "source": [
    "## OLD & BAD\n",
    "# rm -rf ./* > /dev/null\n",
    "# command mv -f * \"$trash_dir\"\n",
    "echo \"Done\""
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
