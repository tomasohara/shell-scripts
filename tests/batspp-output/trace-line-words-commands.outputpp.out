1..17
not ok 1 test-1
# (in test file batspp-output/trace-line-words-commands.outputpp, line 25)
#   `[ "$(test-1-actual)" == "$(test-1-expected)" ]' failed
# ========== echo $PS1
#  ==========
#
# ========= [PEXP\[\]ECT_PROMPT> =========
# [PEXP\[\]ECT_PROMPT>
# ============================
not ok 2 test-3
# (in test file batspp-output/trace-line-words-commands.outputpp, line 59)
#   `[ "$(test-3-actual)" == "$(test-3-expected)" ]' failed
# ========== unalias -a
#  ==========
# ========= $ alias | wc -l\n$ for f in ; do unset -f ; done\n$ typeset -f | egrep '^\w+' | wc -l\n0\n0 =========
# $ alias | wc -l
# $ for f in bats_abort_timeout_countdown ()
# bats_add_debug_exclude_path ()
# bats_all_in ()
# bats_any_in ()
# bats_append_arrays_as_args ()
# bats_approx_realpath ()
# bats_binary_search ()
# bats_capture_stack_trace ()
# bats_check_status_from_trap ()
# bats_check_valid_version ()
# bats_create_test_tmpdirs ()
# bats_debug_trap ()
# bats_emit_trace ()
# bats_error_trap ()
# bats_evaluate_preprocessed_source ()
# bats_execute ()
# bats_exit_trap ()
# bats_extract_line ()
# bats_format_file_line_reference ()
# bats_format_file_line_reference_colon ()
# bats_format_file_line_reference_comma_line ()
# bats_format_file_line_reference_uri ()
# bats_frame_filename ()
# bats_frame_function ()
# bats_frame_lineno ()
# bats_generate_warning ()
# bats_get_child_processes_of ()
# bats_get_failure_stack_trace ()
# bats_internal_load ()
# bats_interrupt_trap ()
# bats_interrupt_trap_in_run ()
# bats_kill_childprocesses_of ()
# bats_load_library ()
# bats_load_library_safe ()
# bats_load_safe ()
# bats_merge_stdout_and_stderr ()
# bats_normalize_windows_dir_path ()
# bats_perform_test ()
# bats_prefix_lines_for_tap_output ()
# bats_preprocess_source ()
# bats_print_failed_command ()
# bats_print_stack_trace ()
# bats_quote_code ()
# bats_readlinkf ()
# declare -fx bats_readlinkf
# bats_redirect_stderr_into_file ()
# bats_replace_filename ()
# bats_require_minimum_version ()
# bats_separate_lines ()
# bats_setup_tracing ()
# bats_should_retry_test ()
# bats_sort ()
# bats_start_timeout_countdown ()
# bats_strip_string ()
# bats_teardown_trap ()
# bats_test_begin ()
# bats_test_function ()
# bats_timeout_trap ()
# bats_trim ()
# bats_trim_filename ()
# bats_version_lt ()
# bats_warn_minimum_guaranteed_version ()
# find_in_bats_lib_path ()
# get_mills_since_epoch ()
# load ()
# run ()
# setup ()
# skip ()
# teardown ()
# test-1-actual ()
# test-1-expected ()
# END_EXPECTED
# test-10-actual ()
# test-10-expected ()
# txt	4
# END_EXPECTED
# test-11-actual ()
# test-11-expected ()
# 2	Hi Mom,
# 6	This is the use of printf
# 11	I can use a backslash n to start a new line
# 1	1
# 1	2
# 1	3
# END_EXPECTED
# test-12-actual ()
# test-12-expected ()
# 7	Hi Mom,
# 25	This is the use of printf
# 43	I can use a backslash n to start a new line
# 1	1
# 1	2
# 0	3
# END_EXPECTED
# test-13-actual ()
# test-13-expected ()
# 82	Hi Mom,
# This is the use of printf
# I can use a backslash n to start a new line
# 1
# 2
# 3
# END_EXPECTED
# test-14-actual ()
# test-14-expected ()
# 2	Hi Mom,
# 6	This is the use of printf
# 11	I can use a backslash n to start a new line
# 1	1
# 1	2
# 1	3
# END_EXPECTED
# test-15-actual ()
# test-15-expected ()
# END_EXPECTED
# test-16-actual ()
# test-16-expected ()
# END_EXPECTED
# test-17-actual ()
# test-17-expected ()
# END_EXPECTED
# test-18-actual ()
# test-18-expected ()
# Done
# END_EXPECTED
# test-3-actual ()
# test-3-expected ()
# 0
# 0
# END_EXPECTED
# test-4-actual ()
# test-4-expected ()
# 0
# END_EXPECTED
# test-5-actual ()
# test-5-expected ()
# END_EXPECTED
# test-6-actual ()
# test-6-expected ()
# 1
# 0
# END_EXPECTED
# test-7-actual ()
# test-7-expected ()
# END_EXPECTED
# test-8-actual ()
# test-8-expected ()
# END_EXPECTED
# test-9-actual ()
# test-9-expected ()
# END_EXPECTED
# test_test-2d1 ()
# test_test-2d10 ()
# test_test-2d11 ()
# 6	This is the use of printf
# 11	I can use a backslash n to start a new line
# 1	1
# 1	2
# 1	3' "=========";
# test_test-2d12 ()
# 25	This is the use of printf
# 43	I can use a backslash n to start a new line
# 1	1
# 1	2
# 0	3' "=========";
# test_test-2d13 ()
# This is the use of printf
# I can use a backslash n to start a new line
# 1
# 2
# 3' "=========";
# test_test-2d14 ()
# 6	This is the use of printf
# 11	I can use a backslash n to start a new line
# 1	1
# 1	2
# 1	3' "=========";
# test_test-2d15 ()
# test_test-2d16 ()
# test_test-2d17 ()
# test_test-2d18 ()
# Done' "=========";
# test_test-2d3 ()
# test_test-2d4 ()
# 0' "=========";
# test_test-2d5 ()
# test_test-2d6 ()
# test_test-2d7 ()
# test_test-2d8 ()
# test_test-2d9 () ; do unset -f ; done
# $ typeset -f | egrep '^\w+' | wc -l
# 0
# 0
# ============================
not ok 3 test-4
# (in test file batspp-output/trace-line-words-commands.outputpp, line 97)
#   `[ "$(test-4-actual)" == "$(test-4-expected)" ]' failed
# ========== TMP=/tmp/test-linewords
#  ==========
# ========= ## NOTE: Source it directly from the ./tests directory.
# $ BIN_DIR=$PWD/..
# $ alias | wc -l
# 0 =========
# ## NOTE: Source it directly from the ./tests directory.
# $ BIN_DIR=/tmp/batspp-70955/test-4/..
# $ alias | wc -l
# 0
# ============================
not ok 4 test-5
# (in test file batspp-output/trace-line-words-commands.outputpp, line 134)
#   `[ "$(test-5-actual)" == "$(test-5-expected)" ]' failed
# ========== temp_dir=$TMP/test-6439
#  ==========
# ========= $ mkdir -p "$temp_dir"
# # TODO: /bin/rm -rvf "$temp_dir"
# $ cd "$temp_dir"
# $ pwd
# #ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)
# $ alias linebr="printf '%*s\n' "${COLUMNS:-$(tput cols)}" '' | tr ' ' -"
# /tmp/test-linewords/test-6439 =========
# $ mkdir -p "/test-6439"
# # TODO: /bin/rm -rvf "/test-6439"
# $ cd "/test-6439"
# $ pwd
# #ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)
# $ alias linebr="printf '%*s\n' "80" '' | tr ' ' -"
# /tmp/test-linewords/test-6439
# ============================
not ok 5 test-6
# (in test file batspp-output/trace-line-words-commands.outputpp, line 174)
#   `[ "$(test-6-actual)" == "$(test-6-expected)" ]' failed
# ========== alias | wc -l
#  ==========
# 0
# ========= # Count functions\n$ typeset -f | egrep '^\w+' | wc -l\n1\n0 =========
# # Count functions
# $ typeset -f | egrep '^\w+' | wc -l
# 1
# 0
# ============================
not ok 6 test-7
# (in test file batspp-output/trace-line-words-commands.outputpp, line 211)
#   `[ "$(test-7-actual)" == "$(test-7-expected)" ]' failed
# ========== alias testnum="sed -r "s/[0-9]/N/g""
#  ==========
# ========= $ alias testuser="sed -r "s/"$USER"+/userxf333/g"" =========
# $ alias testuser="sed -r "s/"aveey"+/userxf333/g""
# ============================
not ok 7 test-8
# (from function `test-8-actual' in file batspp-output/trace-line-words-commands.outputpp, line 252,
#  in test file batspp-output/trace-line-words-commands.outputpp, line 239)
#   `test-8-actual ' failed
# ========== source $BIN_DIR/tomohara-aliases.bash
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-line-words-commands.outputpp: line 252: /tomohara-aliases.bash: No such file or directory
not ok 8 test-9
# (in test file batspp-output/trace-line-words-commands.outputpp, line 279)
#   `[ "$(test-9-actual)" == "$(test-9-expected)" ]' failed
# ========== rm -rf ./*
#  ==========
# ========= $ printf "Hi Mom,\nThis is the use of printf\nI can use a backslash n to start a new line\n1\n2\n3" >> abc-test.txt
# $ printf "This is another-test file" >> test2.txt
# $ printf "This is test-file 3" >> test3.txt
# $ printf "This is a test-file 4" >> test4.txt =========
# $ printf "Hi Mom,\nThis is the use of printf\nI can use a backslash n to start a new line\n1\n2\n3" >> abc-test.txt
# $ printf "This is another-test file" >> test2.txt
# $ printf "This is test-file 3" >> test3.txt
# $ printf "This is a test-file 4" >> test4.txt
# ============================
not ok 9 test-10
# (from function `test-10-actual' in file batspp-output/trace-line-words-commands.outputpp, line 323,
#  in test file batspp-output/trace-line-words-commands.outputpp, line 310)
#   `test-10-actual ' failed with status 127
# ========== ls | count-it '\.([^\.]+)$'\n ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-line-words-commands.outputpp: line 323: count-it: command not found
not ok 10 test-11
# (from function `test-11-actual' in file batspp-output/trace-line-words-commands.outputpp, line 357,
#  in test file batspp-output/trace-line-words-commands.outputpp, line 344)
#   `test-11-actual ' failed with status 127
# ========== cat abc-test.txt | line-wc
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-line-words-commands.outputpp: line 357: line-wc: command not found
# cat: abc-test.txt: No such file or directory
not ok 11 test-12
# (from function `test-12-actual' in file batspp-output/trace-line-words-commands.outputpp, line 396,
#  in test file batspp-output/trace-line-words-commands.outputpp, line 383)
#   `test-12-actual ' failed with status 127
# ========== cat abc-test.txt | line-len
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-line-words-commands.outputpp: line 396: line-len: command not found
# cat: abc-test.txt: No such file or directory
not ok 12 test-13
# (from function `test-13-actual' in file batspp-output/trace-line-words-commands.outputpp, line 435,
#  in test file batspp-output/trace-line-words-commands.outputpp, line 422)
#   `test-13-actual ' failed with status 127
# ========== cat abc-test.txt | para-len
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-line-words-commands.outputpp: line 435: para-len: command not found
# cat: abc-test.txt: No such file or directory
not ok 13 test-14
# (from function `test-14-actual' in file batspp-output/trace-line-words-commands.outputpp, line 474,
#  in test file batspp-output/trace-line-words-commands.outputpp, line 461)
#   `test-14-actual ' failed with status 127
# ========== cat abc-test.txt | line-word-len
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-line-words-commands.outputpp: line 474: line-word-len: command not found
# cat: abc-test.txt: No such file or directory
not ok 14 test-15
# (from function `test-15-actual' in file batspp-output/trace-line-words-commands.outputpp, line 513,
#  in test file batspp-output/trace-line-words-commands.outputpp, line 500)
#   `test-15-actual ' failed with status 127
# ========== ls-line-len
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/trace-line-words-commands.outputpp: line 513: ls-line-len: command not found
not ok 15 test-16
# (in test file batspp-output/trace-line-words-commands.outputpp, line 540)
#   `[ "$(test-16-actual)" == "$(test-16-expected)" ]' failed
# ========== # $ cat abc-test.txt | check-class-dist
#  ==========
# ========= # | Can't open : No such file or directory at /home/xaea12/bin/count_it.perl line 171.\n# | WARNING: unexpected distribution for FREQ (all 0) =========
# # | Can't open : No such file or directory at /home/xaea12/bin/count_it.perl line 171.
# # | WARNING: unexpected distribution for FREQ (all 0)
# ============================
not ok 16 test-17
# (in test file batspp-output/trace-line-words-commands.outputpp, line 574)
#   `[ "$(test-17-actual)" == "$(test-17-expected)" ]' failed
# ========== # $ cat abc-test.txt | 2bib
#  ==========
# ========= # | bash: bibitem2bib: command not found
# # | cat: write error: Broken pipe =========
# # | bash: bibitem2bib: command not found
# # | cat: write error: Broken pipe
# ============================
not ok 17 test-18
# (in test file batspp-output/trace-line-words-commands.outputpp, line 608)
#   `[ "$(test-18-actual)" == "$(test-18-expected)" ]' failed
# ========== rm -rf ./* > /dev/null
#  ==========
# ========= $ echo "Done"
# Done =========
# $ echo "Done"
# Done
# ============================
